cmd/storage-rpc-client.go

```diff
@@ -93,15 +93,14 @@ func toStorageErr(err error) error {
 }
 
 // Initialize new rpc client.
-func newRPCClient(ep storageEndPoint) (StorageAPI, error) {
-	// Input validation.
-	if ep.host == "" || ep.port == 0 || ep.path == "" {
+func newPosixRPC(ep *url.URL) (StorageAPI, error) {
```

Comment:
also unrelated to this patch:

```
type networkStorage struct {                                                                                                                                                    
        netAddr   string                                                                                                                                                        
        netPath   string                                                                                                                                                        
        rpcClient *AuthRPCClient                                                                                                                                                
}
```

this and AuthRPCClient can be combined into one struct and call it maybe rpcClient. Because networkStorage is not doing anything extra other than making calls to AuthRPCClient